actions/AuthAction.js

```diff
@@ -77,6 +48,33 @@ async function isValidToken(endpoint, clientId, token) {
   }
 }
 
+/**
+ * Extracts a user access token from the Authorization header
+ *
+ * @param {object} params action input parameters
+ * @returns {Promise<string>} the token string
+ */
+async function getAccessToken(params) {
+  const imsEndpoint = params.IMS_ENDPOINT;
+  const clientId = params.IMS_CLIENT_ID;
+
+  // First, check if a bearer user access token is set
+  if (
+    params.__ow_headers
+    && params.__ow_headers.authorization
+    && params.__ow_headers.authorization.startsWith('Bearer ')
+  ) {
+    const accessToken = params.__ow_headers.authorization.substring('Bearer '.length);
+    // Validate the access token
```

Comment:
Unfortunately, we cannot utilize AppBuilder's [built-in auth functionality](https://github.com/adobe/aem-genai-assistant/pull/282/commits/d8c3829e1b86bfc0e742f916f728945afd2403c6#diff-6491149c0e11feef942d4d9bd31d89fffaaa3ac4475b77c38443a5ea2d24a60aL21-L23) for user access token validation, as it works only within the scope of the org owning the application. It isn't suitable for cases where there is a need to share the application across different orgs. See more details [here](https://cq-dev.slack.com/archives/C013UDBFBGB/p1716438150500019).